e2e/tests/e2e/JavaVertx.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { Editor } from '../../pageobjects/ide/Editor';
+import { TopMenu } from '../../pageobjects/ide/TopMenu';
+import { QuickOpenContainer } from '../../pageobjects/ide/QuickOpenContainer';
+import { Terminal } from '../../pageobjects/ide/Terminal';
+import 'reflect-metadata';
+import { error, Key } from 'selenium-webdriver';
+import * as restClient from 'typed-rest-client/RestClient';
+import { DriverHelper } from '../../utils/DriverHelper';
+
+const workspaceName: string = NameGenerator.generate('wksp-test-', 5);
+const namespace: string = TestConstants.TS_SELENIUM_USERNAME;
+const sampleName: string = 'java-web-vertx';
+const fileFolderPath: string = `${sampleName}/src/main/java/io/vertx/examples/spring`;
+const tabTitle: string = 'SpringExampleRunner.java';
+const codeNavigationClassName: string = 'ApplicationContext.class';
+
+const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
+const loginPage: ICheLoginPage = e2eContainer.get<ICheLoginPage>(TYPES.CheLogin);
+const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+const newWorkspace: NewWorkspace = e2eContainer.get(CLASSES.NewWorkspace);
+const ide: Ide = e2eContainer.get(CLASSES.Ide);
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree);
+const editor: Editor = e2eContainer.get(CLASSES.Editor);
+const topMenu: TopMenu = e2eContainer.get(CLASSES.TopMenu);
+const quickOpenContainer: QuickOpenContainer = e2eContainer.get(CLASSES.QuickOpenContainer);
+const terminal: Terminal = e2eContainer.get(CLASSES.Terminal);
+
+suite('RhChe E2E Java Vert.x test', async () => {
+    suite('Login and wait dashboard', async () => {
+        test('Login', async () => {
+            console.log('URL: ' + TestConstants.TS_SELENIUM_BASE_URL);
+            await driverHelper.navigateToUrl(TestConstants.TS_SELENIUM_BASE_URL);
+            await loginPage.login();
+        });
+    });
+
+    suite('Create Java Vert.x workspace ' + workspaceName, async () => {
+        test('Open \'New Workspace\' page', async () => {
+            await newWorkspace.openPageByUI();
+        });
+
+        test('Create and open workspace', async () => {
+            await newWorkspace.createAndOpenWorkspace(workspaceName, 'Java Vert.x');
+        });
+    });
+
+    suite('Work with IDE', async () => {
+        test('Wait IDE availability', async () => {
+            await ide.waitWorkspaceAndIde(namespace, workspaceName);
+        });
+
+        test('Open project tree container', async () => {
+            await projectTree.openProjectTreeContainer();
+        });
+
+        test('Wait project imported', async () => {
+            try {
+                await projectTree.waitProjectImported(sampleName, 'src');
+            } catch (err) {
+                if (!(err instanceof error.TimeoutError)) {
+                    throw err;
+                }
+                var rest: restClient.RestClient = new restClient.RestClient('user-agent');
```

Comment:
Why do we use var?

Reply:
Seems like we don't need this try/catch block at all. I'll remove it.

Reply:
Changes pushed. Thank you for valuable comment!